Publish to AWS S3 occasionally fails using 22.06.1-edge and 22.07.1-edge

## Bug report 

### Expected behavior and actual behavior

Hi all 👋. We're running our NextFlow workflows on AWS Batch, using AWS S3 as our `publishDir` for all processes. With NextFlow v22.04.4, the workflows complete as expected and the result files are in their appropriate `publishDir`.

When using NextFlow v22.06.1-edge or v22.07.1-edge, our processes sometimes fail to transfer the output files to the S3 `publishDir` and the workflow only fails when using `failOnError: true` .These files aren't particularly large (ranging from 0.1--3 GB).

### Steps to reproduce the problem

Here's the minimal reproducible example I could come up with.

**main.nf** - `sayHello` creates a specified number of dummy files:
```nextflow
nextflow.enable.dsl=2


process sayHello {
    publishDir params.publishDir, mode: 'copy', failOnError: true

    input:
        val cheers

    output:
        path "${cheers}*.txt"

    shell:
    '''
    for i in {1..!{params.nFiles}}; do
        truncate -s !{params.fileSize}M !{cheers}${i}.txt
    done
    '''
}

workflow {
    channel.of('Ciao', 'Hello', 'Hola') | sayHello
}
```

**nextflow.config*** - the configuration file:
```nextflow
params.publishDir = '' // The publishDir
params.fileSize = 10  // The size file to create in MiB
params.nFiles = 10     // The number of files to create

docker.enabled = true

profiles {
    standard {
        process {
            executor = 'local'
            container = 'amazonlinux:latest'
        }
    }

    aws {
        process {
            executor = 'awsbatch'
            container = 'amazonlinux:latest'
            queue =  // Replace with your queue
            memory = 4.GB
            cpus = 1
        }
    }
}

aws {
    // Your AWS config
}
```

When run locally, with a local `publishDir` this workflow completes successfully:
```sh
nextflow run main.nf -profile standard --publishDir 'test'
```

When run locally, with an S3 `publishDir` this workflow completes successfully:
```sh
nextflow run main.nf \
    -profile standard \
    --publishDir 's3://data-pipeline-experiment-bucket/test-run-s3'
```

When run on AWS Batch, with an S3 `publishDir` this workflow completes successfully using NextFlow v22.04.4, but fails with NextFlow v22.06.1-edge and v22.07.1-edge:
```sh
nextflow run main.nf \
	-profile aws \
	-work-dir 's3://terraform-nextflow-work-bucket/work' \
	--publishDir 's3://data-pipeline-experiment-bucket/test-run-alls3' \
```
### Program output 

```
Jul-13 15:14:25.221 [main] DEBUG nextflow.cli.Launcher - $> nextflow run -c talus.config main.nf -profile aws -work-dir 's3://terraform-nextflow-work-bucket/work' --publishDir 's3://data-pipeline-experiment-bucket/test-run-alls3'
Jul-13 15:14:25.267 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 22.07.1-edge
Jul-13 15:14:25.277 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/Users/wfondrie/.nextflow/plugins; core-plugins: nf-amazon@1.8.1,nf-azure@0.13.3,nf-codecommit@0.1.2,nf-console@1.0.3,nf-ga4gh@1.0.3,nf-google@1.3.0,nf-tower@1.4.0
Jul-13 15:14:25.284 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Enabled plugins: []
Jul-13 15:14:25.284 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Disabled plugins: []
Jul-13 15:14:25.286 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode
Jul-13 15:14:25.292 [main] INFO  org.pf4j.AbstractPluginManager - No plugins
Jul-13 15:14:25.300 [main] DEBUG nextflow.config.ConfigBuilder - Found config local: /Users/wfondrie/scratch/nf-issue/nextflow.config
Jul-13 15:14:25.301 [main] DEBUG nextflow.config.ConfigBuilder - User config file: /Users/wfondrie/scratch/nf-issue/talus.config
Jul-13 15:14:25.301 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/wfondrie/scratch/nf-issue/nextflow.config
Jul-13 15:14:25.301 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/wfondrie/scratch/nf-issue/talus.config
Jul-13 15:14:25.314 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `aws`
Jul-13 15:14:25.627 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `aws`
Jul-13 15:14:25.645 [main] DEBUG nextflow.config.ConfigBuilder - Available config profiles: [standard, aws]
Jul-13 15:14:25.659 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 from script declararion
Jul-13 15:14:25.673 [main] INFO  nextflow.cli.CmdRun - Launching `main.nf` [silly_cantor] DSL2 - revision: 187074bf25
Jul-13 15:14:25.673 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[nf-amazon@1.8.1]
Jul-13 15:14:25.673 [main] DEBUG nextflow.plugin.PluginUpdater - Installing plugin nf-amazon version: 1.8.1
Jul-13 15:14:25.676 [main] INFO  nextflow.plugin.PluginUpdater - Downloading plugin nf-amazon@1.8.1
Jul-13 15:14:36.140 [main] INFO  org.pf4j.util.FileUtils - Expanded plugin zip 'nf-amazon-1.8.1.zip' in 'nf-amazon-1.8.1'
Jul-13 15:14:36.149 [main] INFO  org.pf4j.AbstractPluginManager - Plugin 'nf-amazon@1.8.1' resolved
Jul-13 15:14:36.149 [main] INFO  org.pf4j.AbstractPluginManager - Start plugin 'nf-amazon@1.8.1'
Jul-13 15:14:36.162 [main] DEBUG nextflow.plugin.BasePlugin - Plugin started nf-amazon@1.8.1
Jul-13 15:14:36.176 [main] DEBUG nextflow.file.FileHelper - > Added 'S3FileSystemProvider' to list of installed providers [s3]
Jul-13 15:14:36.213 [main] DEBUG nextflow.Session - Session uuid: 1fbf3a4f-01fc-42a1-b9bb-15aca4072458
Jul-13 15:14:36.213 [main] DEBUG nextflow.Session - Run name: silly_cantor
Jul-13 15:14:36.213 [main] DEBUG nextflow.Session - Executor pool size: 8
Jul-13 15:14:36.226 [main] DEBUG nextflow.file.FileHelper - Creating a file system instance for provider: S3FileSystemProvider
Jul-13 15:14:36.232 [main] DEBUG nextflow.Global - Using AWS credential defined in `default` section in file: /Users/wfondrie/.aws/credentials
Jul-13 15:14:36.236 [main] DEBUG nextflow.file.FileHelper - AWS S3 config details: {secret_key=qV3tln.., region=us-west-2, storage_encryption=AES256, max_error_retry=5, access_key=AKIAZB..}
Jul-13 15:14:36.448 [main] DEBUG com.upplication.s3fs.AmazonS3Client - Setting S3 SSE storage encryption algorithm=AES256
Jul-13 15:14:36.449 [main] DEBUG c.u.s3fs.S3FileSystemProvider - Using S3 multi-part downloader
Jul-13 15:14:36.451 [main] DEBUG c.u.s3fs.ng.S3ParallelDownload - Creating S3 download thread pool: workers=10; chunkSize=10 MB; queueSize=10000; max-mem=1 GB; maxAttempts=5; maxDelay=1m 30s; pool-capacity=103
Jul-13 15:14:36.488 [main] DEBUG nextflow.cli.CmdRun - 
  Version: 22.07.1-edge build 5726
  Created: 13-07-2022 10:53 UTC (03:53 PDT)
  System: Mac OS X 12.4
  Runtime: Groovy 3.0.10 on OpenJDK 64-Bit Server VM 11.0.15+10-LTS
  Encoding: UTF-8 (UTF-8)
  Process: 47418@wills-mbp.lan [192.168.86.129]
  CPUs: 8 - Mem: 8 GB (123.8 MB) - Swap: 3 GB (1.2 GB)
Jul-13 15:14:36.959 [main] DEBUG nextflow.Session - Work-dir: s3://terraform-nextflow-work-bucket/work [Mac OS X]
Jul-13 15:14:36.959 [main] DEBUG nextflow.Session - Script base path does not exist or is not a directory: /Users/wfondrie/scratch/nf-issue/bin
Jul-13 15:14:36.978 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[AwsBatchExecutor]
Jul-13 15:14:36.985 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory
Jul-13 15:14:36.997 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory
Jul-13 15:14:37.004 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000
Jul-13 15:14:37.043 [main] DEBUG nextflow.Session - Session start invoked
Jul-13 15:14:37.166 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution
Jul-13 15:14:37.216 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: awsbatch
Jul-13 15:14:37.216 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'awsbatch'
Jul-13 15:14:37.217 [main] DEBUG nextflow.executor.Executor - [warm up] executor > awsbatch
Jul-13 15:14:37.224 [main] DEBUG nextflow.util.ThrottlingExecutor - Creating throttling executor with opts: nextflow.util.ThrottlingExecutor$Options(poolName:AWSBatch-executor, limiter:RateLimiter[stableRate=50.0qps], poolSize:40, maxPoolSize:40, queueSize:5000, maxRetries:10, keepAlive:1m, autoThrottle:true, errorBurstDelay:1s, rampUpInterval:100, rampUpFactor:1.2, rampUpMaxRate:1.7976931348623157E308, backOffFactor:2.0, backOffMinRate:0.0166666667, retryDelay:1s)
Jul-13 15:14:37.227 [main] DEBUG nextflow.util.ThrottlingExecutor - Creating throttling executor with opts: nextflow.util.ThrottlingExecutor$Options(poolName:AWSBatch-reaper, limiter:RateLimiter[stableRate=50.0qps], poolSize:40, maxPoolSize:40, queueSize:5000, maxRetries:10, keepAlive:1m, autoThrottle:true, errorBurstDelay:1s, rampUpInterval:100, rampUpFactor:1.2, rampUpMaxRate:1.7976931348623157E308, backOffFactor:2.0, backOffMinRate:0.0166666667, retryDelay:1s)
Jul-13 15:14:37.228 [main] DEBUG n.cloud.aws.batch.AwsBatchExecutor - Creating parallel monitor for executor 'awsbatch' > pollInterval=10s; dumpInterval=5m
Jul-13 15:14:37.245 [main] DEBUG nextflow.Global - Using AWS credential defined in `default` section in file: /Users/wfondrie/.aws/credentials
Jul-13 15:14:37.269 [main] DEBUG n.cloud.aws.batch.AwsBatchExecutor - [AWS BATCH] Executor options=AwsOptions(cliPath:/home/ec2-user/bin/aws, storageClass:null, storageEncryption:AES256, storageKmsKeyId:null, remoteBinDir:null, region:us-west-2, maxParallelTransfers:4, maxTransferAttempts:5, delayBetweenAttempts:10s, retryMode:standard, maxSpotAttempts:5, debug:null, fetchInstanceType:false, jobRole:arn:aws:iam::622568582929:role/terraform-nextflow-batch-job-role, volumes:[], awsCli:/home/ec2-user/bin/aws --region us-west-2)
Jul-13 15:14:37.305 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: sayHello
Jul-13 15:14:37.306 [main] DEBUG nextflow.Session - Ignite dataflow network (2)
Jul-13 15:14:37.306 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > sayHello
Jul-13 15:14:37.306 [main] DEBUG nextflow.script.ScriptRunner - > Await termination 
Jul-13 15:14:37.306 [main] DEBUG nextflow.Session - Session await
Jul-13 15:14:38.000 [AWSBatch-executor-1] DEBUG n.c.aws.batch.AwsBatchTaskHandler - [AWS BATCH] [AWS BATCH] Found job definition name=nf-amazonlinux-latest:1; container=amazonlinux:latest
Jul-13 15:14:38.138 [AWSBatch-executor-2] DEBUG n.c.aws.batch.AwsBatchTaskHandler - [AWS BATCH] submitted > job=353b3fc6-33cb-466d-92fd-c20d4339468e; work-dir=s3://terraform-nextflow-work-bucket/work/63/7357b3c8f29b2efe7d55375420b3d1
Jul-13 15:14:38.138 [AWSBatch-executor-2] INFO  nextflow.Session - [63/7357b3] Submitted process > sayHello (2)
Jul-13 15:14:38.198 [AWSBatch-executor-1] DEBUG n.c.aws.batch.AwsBatchTaskHandler - [AWS BATCH] submitted > job=d738dfd7-4ef3-47e3-8d60-5b280d8d528e; work-dir=s3://terraform-nextflow-work-bucket/work/c2/adb1d9cb89fac7602d2428f4c2cda6
Jul-13 15:14:38.198 [AWSBatch-executor-1] INFO  nextflow.Session - [c2/adb1d9] Submitted process > sayHello (1)
Jul-13 15:14:38.238 [AWSBatch-executor-3] DEBUG n.c.aws.batch.AwsBatchTaskHandler - [AWS BATCH] submitted > job=b4e8bf4b-4658-4b71-86b8-45e75c20a9b9; work-dir=s3://terraform-nextflow-work-bucket/work/d2/ace1fd450aa27bf0d705afa0ccc210
Jul-13 15:14:38.238 [AWSBatch-executor-3] INFO  nextflow.Session - [d2/ace1fd] Submitted process > sayHello (3)
Jul-13 15:17:27.462 [Task monitor] DEBUG c.u.s3fs.S3FileSystemProvider - S3 parallel download with direct buffer: s3://terraform-nextflow-work-bucket/work/63/7357b3c8f29b2efe7d55375420b3d1/.exitcode
Jul-13 15:17:27.879 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 2; name: sayHello (2); status: COMPLETED; exit: 0; error: -; workDir: s3://terraform-nextflow-work-bucket/work/63/7357b3c8f29b2efe7d55375420b3d1]
Jul-13 15:17:28.537 [Task monitor] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=4; maxSize=4; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false
Jul-13 15:17:28.546 [Task monitor] DEBUG c.u.s3fs.S3FileSystemProvider - S3 parallel download with direct buffer: s3://terraform-nextflow-work-bucket/work/63/7357b3c8f29b2efe7d55375420b3d1/.command.trace
Jul-13 15:17:28.730 [FileTransfer-4] WARN  nextflow.processor.PublishDir - Failed to publish file: s3://terraform-nextflow-work-bucket/work/63/7357b3c8f29b2efe7d55375420b3d1/Hello4.txt; to: s3://data-pipeline-experiment-bucket/test-run-alls3/Hello4.txt [copy] -- See log file for details
java.lang.NullPointerException: null
	at com.upplication.s3fs.AmazonS3Client.copyObject(AmazonS3Client.java:196)
	at com.upplication.s3fs.S3FileSystemProvider.copy(S3FileSystemProvider.java:573)
	at nextflow.file.FileHelper.copyPath(FileHelper.groovy:934)
	at nextflow.processor.PublishDir.processFileImpl(PublishDir.groovy:436)
	at nextflow.processor.PublishDir.processFile(PublishDir.groovy:347)
	at nextflow.processor.PublishDir.safeProcessFile(PublishDir.groovy:329)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
	at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:1029)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:1012)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethodSafe(InvokerHelper.java:101)
	at nextflow.processor.PublishDir$_apply1_closure1.doCall(PublishDir.groovy:302)
	at nextflow.processor.PublishDir$_apply1_closure1.call(PublishDir.groovy)
	at groovy.lang.Closure.run(Closure.java:493)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Jul-13 15:17:28.732 [FileTransfer-4] DEBUG nextflow.Session - Session aborted -- Cause: java.lang.NullPointerException
Jul-13 15:17:28.741 [Task monitor] DEBUG c.u.s3fs.S3FileSystemProvider - S3 parallel download with direct buffer: s3://terraform-nextflow-work-bucket/work/c2/adb1d9cb89fac7602d2428f4c2cda6/.exitcode
Jul-13 15:17:28.744 [main] DEBUG nextflow.Session - Session await > all process finished
Jul-13 15:17:28.745 [main] DEBUG nextflow.Session - Session await > all barriers passed
Jul-13 15:17:28.750 [main] WARN  n.processor.TaskPollingMonitor - Killing running tasks (2)
Jul-13 15:17:28.761 [main] DEBUG nextflow.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=1; failedCount=0; ignoredCount=0; cachedCount=0; pendingCount=0; submittedCount=1; runningCount=-1; retriesCount=0; abortedCount=2; succeedDuration=30.5s; failedDuration=0ms; cachedDuration=0ms;loadCpus=-1; loadMemory=0; peakRunning=2; peakCpus=2; peakMemory=8 GB; ]
Jul-13 15:17:28.800 [FileTransfer-4] WARN  nextflow.processor.PublishDir - Failed to publish file: s3://terraform-nextflow-work-bucket/work/63/7357b3c8f29b2efe7d55375420b3d1/Hello7.txt; to: s3://data-pipeline-experiment-bucket/test-run-alls3/Hello7.txt [copy] -- See log file for details
com.amazonaws.SdkClientException: Failed to sanitize XML document destined for handler class com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$ListBucketHandler
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.sanitizeXmlDocument(XmlResponsesSaxParser.java:239)
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.parseListBucketObjectsResponse(XmlResponsesSaxParser.java:338)
	at com.amazonaws.services.s3.model.transform.Unmarshallers$ListObjectsUnmarshaller.unmarshall(Unmarshallers.java:118)
	at com.amazonaws.services.s3.model.transform.Unmarshallers$ListObjectsUnmarshaller.unmarshall(Unmarshallers.java:107)
	at com.amazonaws.services.s3.internal.S3XmlResponseHandler.handle(S3XmlResponseHandler.java:62)
	at com.amazonaws.services.s3.internal.S3XmlResponseHandler.handle(S3XmlResponseHandler.java:31)
	at com.amazonaws.http.response.AwsResponseHandlerAdapter.handle(AwsResponseHandlerAdapter.java:69)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleResponse(AmazonHttpClient.java:1777)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleSuccessResponse(AmazonHttpClient.java:1474)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1381)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1154)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:811)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:779)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:753)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:713)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:695)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:559)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:539)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5453)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5400)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5394)
	at com.amazonaws.services.s3.AmazonS3Client.listObjects(AmazonS3Client.java:928)
	at com.upplication.s3fs.AmazonS3Client.listObjects(AmazonS3Client.java:135)
	at com.upplication.s3fs.util.S3ObjectSummaryLookup.lookup(S3ObjectSummaryLookup.java:113)
	at com.upplication.s3fs.S3FileSystemProvider.readAttr0(S3FileSystemProvider.java:700)
	at com.upplication.s3fs.S3FileSystemProvider.readAttributes(S3FileSystemProvider.java:693)
	at java.base/java.nio.file.Files.readAttributes(Files.java:1764)
	at java.base/java.nio.file.Files.isDirectory(Files.java:2235)
	at nextflow.file.FileHelper.copyPath(FileHelper.groovy:930)
	at nextflow.processor.PublishDir.processFileImpl(PublishDir.groovy:436)
	at nextflow.processor.PublishDir.processFile(PublishDir.groovy:347)
	at nextflow.processor.PublishDir.safeProcessFile(PublishDir.groovy:329)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
	at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:1029)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:1012)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethodSafe(InvokerHelper.java:101)
	at nextflow.processor.PublishDir$_apply1_closure1.doCall(PublishDir.groovy:302)
	at nextflow.processor.PublishDir$_apply1_closure1.call(PublishDir.groovy)
	at groovy.lang.Closure.run(Closure.java:493)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.amazonaws.AbortedException: 
	at com.amazonaws.internal.SdkFilterInputStream.abortIfNeeded(SdkFilterInputStream.java:61)
	at com.amazonaws.internal.SdkFilterInputStream.read(SdkFilterInputStream.java:89)
	at com.amazonaws.event.ProgressInputStream.read(ProgressInputStream.java:180)
	at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
	at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
	at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	at java.base/java.io.InputStreamReader.read(InputStreamReader.java:181)
	at java.base/java.io.BufferedReader.read1(BufferedReader.java:210)
	at java.base/java.io.BufferedReader.read(BufferedReader.java:287)
	at java.base/java.io.Reader.read(Reader.java:229)
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.sanitizeXmlDocument(XmlResponsesSaxParser.java:211)
	... 50 common frames omitted
Jul-13 15:17:28.803 [FileTransfer-3] WARN  nextflow.processor.PublishDir - Failed to publish file: s3://terraform-nextflow-work-bucket/work/63/7357b3c8f29b2efe7d55375420b3d1/Hello3.txt; to: s3://data-pipeline-experiment-bucket/test-run-alls3/Hello3.txt [copy] -- See log file for details
com.amazonaws.SdkClientException: Failed to sanitize XML document destined for handler class com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$ListBucketHandler
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.sanitizeXmlDocument(XmlResponsesSaxParser.java:239)
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.parseListBucketObjectsResponse(XmlResponsesSaxParser.java:338)
	at com.amazonaws.services.s3.model.transform.Unmarshallers$ListObjectsUnmarshaller.unmarshall(Unmarshallers.java:118)
	at com.amazonaws.services.s3.model.transform.Unmarshallers$ListObjectsUnmarshaller.unmarshall(Unmarshallers.java:107)
	at com.amazonaws.services.s3.internal.S3XmlResponseHandler.handle(S3XmlResponseHandler.java:62)
	at com.amazonaws.services.s3.internal.S3XmlResponseHandler.handle(S3XmlResponseHandler.java:31)
	at com.amazonaws.http.response.AwsResponseHandlerAdapter.handle(AwsResponseHandlerAdapter.java:69)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleResponse(AmazonHttpClient.java:1777)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleSuccessResponse(AmazonHttpClient.java:1474)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1381)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1154)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:811)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:779)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:753)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:713)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:695)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:559)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:539)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5453)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5400)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5394)
	at com.amazonaws.services.s3.AmazonS3Client.listObjects(AmazonS3Client.java:928)
	at com.upplication.s3fs.AmazonS3Client.listObjects(AmazonS3Client.java:135)
	at com.upplication.s3fs.util.S3ObjectSummaryLookup.lookup(S3ObjectSummaryLookup.java:113)
	at com.upplication.s3fs.S3FileSystemProvider.readAttr0(S3FileSystemProvider.java:700)
	at com.upplication.s3fs.S3FileSystemProvider.readAttributes(S3FileSystemProvider.java:693)
	at java.base/java.nio.file.Files.readAttributes(Files.java:1764)
	at java.base/java.nio.file.Files.isDirectory(Files.java:2235)
	at nextflow.file.FileHelper.copyPath(FileHelper.groovy:930)
	at nextflow.processor.PublishDir.processFileImpl(PublishDir.groovy:436)
	at nextflow.processor.PublishDir.processFile(PublishDir.groovy:347)
	at nextflow.processor.PublishDir.safeProcessFile(PublishDir.groovy:329)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
	at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:1029)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:1012)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethodSafe(InvokerHelper.java:101)
	at nextflow.processor.PublishDir$_apply1_closure1.doCall(PublishDir.groovy:302)
	at nextflow.processor.PublishDir$_apply1_closure1.call(PublishDir.groovy)
	at groovy.lang.Closure.run(Closure.java:493)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.amazonaws.AbortedException: 
	at com.amazonaws.internal.SdkFilterInputStream.abortIfNeeded(SdkFilterInputStream.java:61)
	at com.amazonaws.internal.SdkFilterInputStream.read(SdkFilterInputStream.java:89)
	at com.amazonaws.event.ProgressInputStream.read(ProgressInputStream.java:180)
	at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
	at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
	at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	at java.base/java.io.InputStreamReader.read(InputStreamReader.java:181)
	at java.base/java.io.BufferedReader.read1(BufferedReader.java:210)
	at java.base/java.io.BufferedReader.read(BufferedReader.java:287)
	at java.base/java.io.Reader.read(Reader.java:229)
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.sanitizeXmlDocument(XmlResponsesSaxParser.java:211)
	... 50 common frames omitted
Jul-13 15:17:28.814 [FileTransfer-2] WARN  nextflow.processor.PublishDir - Failed to publish file: s3://terraform-nextflow-work-bucket/work/63/7357b3c8f29b2efe7d55375420b3d1/Hello2.txt; to: s3://data-pipeline-experiment-bucket/test-run-alls3/Hello2.txt [copy] -- See log file for details
com.amazonaws.SdkClientException: Failed to sanitize XML document destined for handler class com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$ListBucketHandler
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.sanitizeXmlDocument(XmlResponsesSaxParser.java:239)
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.parseListBucketObjectsResponse(XmlResponsesSaxParser.java:338)
	at com.amazonaws.services.s3.model.transform.Unmarshallers$ListObjectsUnmarshaller.unmarshall(Unmarshallers.java:118)
	at com.amazonaws.services.s3.model.transform.Unmarshallers$ListObjectsUnmarshaller.unmarshall(Unmarshallers.java:107)
	at com.amazonaws.services.s3.internal.S3XmlResponseHandler.handle(S3XmlResponseHandler.java:62)
	at com.amazonaws.services.s3.internal.S3XmlResponseHandler.handle(S3XmlResponseHandler.java:31)
	at com.amazonaws.http.response.AwsResponseHandlerAdapter.handle(AwsResponseHandlerAdapter.java:69)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleResponse(AmazonHttpClient.java:1777)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleSuccessResponse(AmazonHttpClient.java:1474)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1381)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1154)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:811)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:779)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:753)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:713)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:695)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:559)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:539)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5453)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5400)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5394)
	at com.amazonaws.services.s3.AmazonS3Client.listObjects(AmazonS3Client.java:928)
	at com.upplication.s3fs.AmazonS3Client.listObjects(AmazonS3Client.java:135)
	at com.upplication.s3fs.util.S3ObjectSummaryLookup.lookup(S3ObjectSummaryLookup.java:113)
	at com.upplication.s3fs.S3FileSystemProvider.readAttr0(S3FileSystemProvider.java:700)
	at com.upplication.s3fs.S3FileSystemProvider.readAttributes(S3FileSystemProvider.java:693)
	at java.base/java.nio.file.Files.readAttributes(Files.java:1764)
	at java.base/java.nio.file.Files.isDirectory(Files.java:2235)
	at nextflow.file.FileHelper.copyPath(FileHelper.groovy:930)
	at nextflow.processor.PublishDir.processFileImpl(PublishDir.groovy:436)
	at nextflow.processor.PublishDir.processFile(PublishDir.groovy:347)
	at nextflow.processor.PublishDir.safeProcessFile(PublishDir.groovy:329)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
	at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:1029)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:1012)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethodSafe(InvokerHelper.java:101)
	at nextflow.processor.PublishDir$_apply1_closure1.doCall(PublishDir.groovy:302)
	at nextflow.processor.PublishDir$_apply1_closure1.call(PublishDir.groovy)
	at groovy.lang.Closure.run(Closure.java:493)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.amazonaws.AbortedException: 
	at com.amazonaws.internal.SdkFilterInputStream.abortIfNeeded(SdkFilterInputStream.java:61)
	at com.amazonaws.internal.SdkFilterInputStream.read(SdkFilterInputStream.java:89)
	at com.amazonaws.event.ProgressInputStream.read(ProgressInputStream.java:180)
	at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
	at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
	at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	at java.base/java.io.InputStreamReader.read(InputStreamReader.java:181)
	at java.base/java.io.BufferedReader.read1(BufferedReader.java:210)
	at java.base/java.io.BufferedReader.read(BufferedReader.java:287)
	at java.base/java.io.Reader.read(Reader.java:229)
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.sanitizeXmlDocument(XmlResponsesSaxParser.java:211)
	... 50 common frames omitted
Jul-13 15:17:28.889 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 1; name: sayHello (1); status: COMPLETED; exit: 0; error: -; workDir: s3://terraform-nextflow-work-bucket/work/c2/adb1d9cb89fac7602d2428f4c2cda6]
Jul-13 15:17:28.914 [FileTransfer-1] WARN  nextflow.processor.PublishDir - Failed to publish file: s3://terraform-nextflow-work-bucket/work/63/7357b3c8f29b2efe7d55375420b3d1/Hello10.txt; to: s3://data-pipeline-experiment-bucket/test-run-alls3/Hello10.txt [copy] -- See log file for details
com.amazonaws.SdkClientException: Failed to sanitize XML document destined for handler class com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser$ListBucketHandler
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.sanitizeXmlDocument(XmlResponsesSaxParser.java:239)
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.parseListBucketObjectsResponse(XmlResponsesSaxParser.java:338)
	at com.amazonaws.services.s3.model.transform.Unmarshallers$ListObjectsUnmarshaller.unmarshall(Unmarshallers.java:118)
	at com.amazonaws.services.s3.model.transform.Unmarshallers$ListObjectsUnmarshaller.unmarshall(Unmarshallers.java:107)
	at com.amazonaws.services.s3.internal.S3XmlResponseHandler.handle(S3XmlResponseHandler.java:62)
	at com.amazonaws.services.s3.internal.S3XmlResponseHandler.handle(S3XmlResponseHandler.java:31)
	at com.amazonaws.http.response.AwsResponseHandlerAdapter.handle(AwsResponseHandlerAdapter.java:69)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleResponse(AmazonHttpClient.java:1777)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleSuccessResponse(AmazonHttpClient.java:1474)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1381)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1154)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:811)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:779)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:753)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:713)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:695)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:559)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:539)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5453)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5400)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5394)
	at com.amazonaws.services.s3.AmazonS3Client.listObjects(AmazonS3Client.java:928)
	at com.upplication.s3fs.AmazonS3Client.listObjects(AmazonS3Client.java:135)
	at com.upplication.s3fs.util.S3ObjectSummaryLookup.lookup(S3ObjectSummaryLookup.java:113)
	at com.upplication.s3fs.S3FileSystemProvider.readAttr0(S3FileSystemProvider.java:700)
	at com.upplication.s3fs.S3FileSystemProvider.readAttributes(S3FileSystemProvider.java:693)
	at java.base/java.nio.file.Files.readAttributes(Files.java:1764)
	at java.base/java.nio.file.Files.isDirectory(Files.java:2235)
	at nextflow.file.FileHelper.copyPath(FileHelper.groovy:930)
	at nextflow.processor.PublishDir.processFileImpl(PublishDir.groovy:436)
	at nextflow.processor.PublishDir.processFile(PublishDir.groovy:347)
	at nextflow.processor.PublishDir.safeProcessFile(PublishDir.groovy:329)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
	at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:1029)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:1012)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethodSafe(InvokerHelper.java:101)
	at nextflow.processor.PublishDir$_apply1_closure1.doCall(PublishDir.groovy:302)
	at nextflow.processor.PublishDir$_apply1_closure1.call(PublishDir.groovy)
	at groovy.lang.Closure.run(Closure.java:493)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: com.amazonaws.AbortedException: 
	at com.amazonaws.internal.SdkFilterInputStream.abortIfNeeded(SdkFilterInputStream.java:61)
	at com.amazonaws.internal.SdkFilterInputStream.read(SdkFilterInputStream.java:89)
	at com.amazonaws.event.ProgressInputStream.read(ProgressInputStream.java:180)
	at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284)
	at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326)
	at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178)
	at java.base/java.io.InputStreamReader.read(InputStreamReader.java:181)
	at java.base/java.io.BufferedReader.read1(BufferedReader.java:210)
	at java.base/java.io.BufferedReader.read(BufferedReader.java:287)
	at java.base/java.io.Reader.read(Reader.java:229)
	at com.amazonaws.services.s3.model.transform.XmlResponsesSaxParser.sanitizeXmlDocument(XmlResponsesSaxParser.java:211)
	... 50 common frames omitted
Jul-13 15:17:28.952 [AWSBatch-reaper-1] DEBUG n.c.aws.batch.AwsBatchTaskHandler - [AWS BATCH] killing job=d738dfd7-4ef3-47e3-8d60-5b280d8d528e; response={}
Jul-13 15:17:29.006 [AWSBatch-reaper-2] DEBUG n.c.aws.batch.AwsBatchTaskHandler - [AWS BATCH] killing job=b4e8bf4b-4658-4b71-86b8-45e75c20a9b9; response={}
Jul-13 15:17:29.013 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done
Jul-13 15:17:29.013 [main] INFO  org.pf4j.AbstractPluginManager - Stop plugin 'nf-amazon@1.8.1'
Jul-13 15:17:29.013 [main] DEBUG nextflow.plugin.BasePlugin - Plugin stopped nf-amazon
Jul-13 15:17:29.013 [main] DEBUG c.u.s3fs.ng.S3ParallelDownload - Shutdown S3 downloader
Jul-13 15:17:29.013 [main] DEBUG c.u.s3fs.ng.S3ParallelDownload - Shutdown S3 downloader - done
Jul-13 15:17:29.024 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye
Jul-13 15:17:29.024 [Task monitor] DEBUG nextflow.plugin.PluginsFacade - Using Default plugins manager
Jul-13 15:17:29.029 [Task monitor] INFO  org.pf4j.DefaultPluginStatusProvider - Enabled plugins: []
Jul-13 15:17:29.029 [Task monitor] INFO  org.pf4j.DefaultPluginStatusProvider - Disabled plugins: []
Jul-13 15:17:29.033 [Task monitor] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode
Jul-13 15:17:29.034 [Task monitor] WARN  nextflow.file.FileHelper - Unable to start plugin 'nf-amazon' required by s3://data-pipeline-experiment-bucket/test-run-alls3
java.lang.NullPointerException: null
	at nextflow.plugin.PluginsFacade.isStarted(PluginsFacade.groovy:322)
	at nextflow.plugin.PluginsFacade.startIfMissing(PluginsFacade.groovy:411)
	at nextflow.plugin.Plugins.startIfMissing(Plugins.groovy:85)
	at nextflow.file.FileHelper.autoStartMissingPlugin(FileHelper.groovy:288)
	at nextflow.file.FileHelper.asPath(FileHelper.groovy:268)
	at nextflow.extension.Bolts.asType(Bolts.groovy:439)
	at jdk.internal.reflect.GeneratedMethodAccessor18.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.runtime.metaclass.ReflectionMetaMethod.invoke(ReflectionMetaMethod.java:54)
	at org.codehaus.groovy.runtime.metaclass.NewInstanceMetaMethod.invoke(NewInstanceMetaMethod.java:54)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
	at org.codehaus.groovy.runtime.InvokerHelper.invokePojoMethod(InvokerHelper.java:1017)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:1008)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodN(ScriptBytecodeAdapter.java:180)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.asType(ScriptBytecodeAdapter.java:603)
	at nextflow.processor.PublishDir.setPath(PublishDir.groovy:120)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
	at org.codehaus.groovy.runtime.metaclass.MultipleSetterProperty.setProperty(MultipleSetterProperty.java:54)
	at groovy.lang.MetaClassImpl.setProperty(MetaClassImpl.java:2801)
	at groovy.lang.MetaClassImpl.setProperty(MetaClassImpl.java:3854)
	at groovy.lang.GroovyObject.setProperty(GroovyObject.java:61)
	at org.codehaus.groovy.runtime.InvokerHelper.setProperty(InvokerHelper.java:217)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.setProperty(ScriptBytecodeAdapter.java:496)
	at nextflow.processor.PublishDir.create(PublishDir.groovy:166)
	at nextflow.processor.TaskConfig.getPublishDir(TaskConfig.groovy:367)
	at nextflow.processor.TaskProcessor.publishOutputs(TaskProcessor.groovy:1278)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)
	at nextflow.processor.TaskProcessor.finalizeTask0(TaskProcessor.groovy:2189)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)
	at nextflow.processor.TaskProcessor.finalizeTask(TaskProcessor.groovy:2160)
	at nextflow.processor.TaskPollingMonitor.checkTaskStatus(TaskPollingMonitor.groovy:619)
	at nextflow.processor.TaskPollingMonitor.checkAllTasks(TaskPollingMonitor.groovy:534)
	at nextflow.processor.TaskPollingMonitor.pollLoop(TaskPollingMonitor.groovy:409)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
	at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:1029)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:1012)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethodSafe(InvokerHelper.java:101)
	at nextflow.processor.TaskPollingMonitor$_start_closure2.doCall(TaskPollingMonitor.groovy:293)
	at nextflow.processor.TaskPollingMonitor$_start_closure2.call(TaskPollingMonitor.groovy)
	at groovy.lang.Closure.run(Closure.java:493)
	at java.base/java.lang.Thread.run(Thread.java:829)
Jul-13 15:17:29.035 [Task monitor] DEBUG nextflow.plugin.PluginsFacade - Using Default plugins manager
Jul-13 15:17:29.037 [Task monitor] DEBUG nextflow.plugin.PluginsFacade - Using Default plugins manager
Jul-13 15:17:29.038 [Task monitor] DEBUG nextflow.plugin.PluginsFacade - Using Default plugins manager
```
### Environment 

* Nextflow version: 22.07.1-edge build 5726
* Java version: Groovy 3.0.10 on OpenJDK 64-Bit Server VM 11.0.15+10-LTS
* Operating system: MacOS (runner), Linux on AWS Batch (jobs)
* Bash version: zsh 5.8.1 (x86_64-apple-darwin21.0)

### Additional context

Not really additional context, but is there a reason why `failOnError` is `false` by default? It took me awhile to figure out what was going on because previously these jobs would complete without any errors.

--- a/plugins/nf-amazon/src/main/com/upplication/s3fs/AmazonS3Client.java
+++ b/plugins/nf-amazon/src/main/com/upplication/s3fs/AmazonS3Client.java
@@ -233,7 +233,8 @@ public class AmazonS3Client {
 			req.withCannedAccessControlList(cannedAcl);
 		}
 		if( storageEncryption != null ) {
-			ObjectMetadata meta = req.getNewObjectMetadata();
+			// getNewObjectMetada returns null if no object metadata has been specified.
+			ObjectMetadata meta = req.getNewObjectMetadata() != null ? req.getNewObjectMetadata() : new ObjectMetadata();
 			meta.setSSEAlgorithm(storageEncryption.toString());
 			req.setNewObjectMetadata(meta);
 		}

